src/logic/calculate.js

```diff
@@ -0,0 +1,20 @@
+import operate from './operate';
+
+const calculate = (obj, buttonName) => {
+  const calcData = { obj };
+  if (buttonName === '+/-') {
+    calcData.total *= -1;
+    calcData.next *= -1;
+  } else if (buttonName === 'AC') {
+    calcData.total = null;
+    calcData.next = null;
+  } else if (buttonName === '=') {
+    if (calcData.total && calcData.next) {
+      calcData.total = operate(calcData.total, calcData.next, buttonName);
+    }
+  } else if (buttonName === '+' || buttonName === '-' || buttonName === 'x' || buttonName === '÷') {
+    calcData.total = operate(calcData.total, calcData.next, buttonName);
+  }
+  return obj;
```

Comment:
- [ ] Please include a condition that covers the buttonName as a number or a decimal(.) operator.